packages/mongodb-rag-core/src/contentStore/Page.ts

```diff
@@ -83,6 +83,21 @@ export type LoadPagesArgs<QueryShape = unknown> = {
   urls?: string[];
 };
 
+export type DeletePagesArgs = {
+    /**
+     Specifies deletion criteria using MQL query operators.
+     To delete all documents in the collection, pass in an empty document ({ }).
+    */
+    filter: {
+      [key: string]: any;
+    };
```

Comment:
i think there's better typescript typing you can do here. think there's a type for `Filter`, if that doesn't work, then `Document` still better than this. 

i'd ask chatgpt for clarification

Reply:
I could set this up to be more similar to the [`deleteEmbeddedContent` args](https://github.com/mongodb/chatbot/pull/556/files#diff-8d82131cb2cd7aa812e94a672fe0a1df4f4230dc736cb962e2d9af88768df78e) the specific keys `deletePages` accepts are defined. As of now there is only one instance of the delete filter and it looks like `{sourceName: {$in: sourcesToDeleteArray}}`, so I can set up `DeletePagesArgs` like this:
```
export type DeletePagesArgs = {
    /**
     Specifies deletion criteria using MQL query operators.
     To delete all documents in the collection, pass in an empty document ({ }).
    */
    filter: {
      dataSources: string[];
      // add more as needed
    };
    /**
    Permanently remove pages from the data store,
    rather than marking them as `"deleted"`.
   */
    permanent?: boolean;
  };
```